librz/include/rz_th.h

```diff
@@ -5,44 +5,39 @@
 #undef _GNU_SOURCE
 #endif
 #define _GNU_SOURCE
-#include "rz_types.h"
+#include <rz_types.h>
+#include <rz_list.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#define RZ_THREAD_POOL_ALL_CORES  (0)
+#define RZ_THREAD_QUEUE_UNLIMITED (0)
+
 typedef enum {
-	RZ_TH_FREED = -1,
-	RZ_TH_STOP = 0,
-	RZ_TH_REPEAT = 1
-} RzThreadFunctionRet;
-#define RZ_TH_FUNCTION(x) RzThreadFunctionRet (*x)(struct rz_th_t *)
+	RZ_TH_STATUS_STOP = 0, ///< To return to terminate the RzThreadFunction execution
+	RZ_TH_STATUS_LOOP = 1, ///< To return to execute multiple times the RzThreadFunction
+} RzThreadStatus;
 
 typedef struct rz_th_sem_t RzThreadSemaphore;
 typedef struct rz_th_lock_t RzThreadLock;
 typedef struct rz_th_cond_t RzThreadCond;
 typedef struct rz_th_t RzThread;
-
-#define RZ_THREAD_POOL_ALL_CORES (0)
-
-typedef struct rz_th_pool_t {
-	size_t size;
-	RzThread **threads;
-} RzThreadPool;
+typedef struct rz_th_pool_t RzThreadPool;
+typedef struct rz_th_queue_t RzThreadQueue;
```

Comment:
Please add some docs on what exactly this queue is. I.e. "a thread-safe queue that can be listened on from multiple threads" for example.
Also please document the functions describing exactly the interesting parts of the expected behavior, e.g. when you pop, will it block and wait or will it return immediately, what happens when multiple threads pop concurrently, when the queue is empty, etc.